### PR TITLE
Gizmo fixes

### LINKED ISSUE #6671: Gizmo sizing bug
Gizmo appears different sizes under different circumstances.

Seems to be related to the `Gizmo._deviceStartSize` value which should just removed completely. (See [here](https://github.com/playcanvas/engine/blob/main/src/extras/gizmo/gizmo.js#L257)).

### LINKED ISSUE #6695: Gizmo is not interactable when close
### Description
Gizmo is not interactable when close

### Steps to Reproduce

1. go to examples https://playcanvas.github.io/#/misc/gizmos
2. get really close to the gizmo
3. it is not possible to interact with the gizmo

https://github.com/playcanvas/engine/assets/100127317/348d1824-5ceb-4d2c-83f8-f6751b47bdf4

### PR DESCRIPTION
**Pull request by kpal81xd**
Fixes #6671 
Fixes #6695 

- Gizmo scale now just dependent on FOV and distance from camera
- Selection ray fired from near plane instead of a distance of 1 from the camera
- Added JSDoc to all methods in gizmo classes 
- Used event names instead of raw strings for callbacks and fire methods
- Used axis constants instead of raw string values

### CODE REVIEW

File: src/extras/gizmo/axis-shapes.js [hunk 1]
```diff
@@ -51,7 +51,7 @@ const SHADER = {
             // store z/w for later use in fragment shader
             vZW = gl_Position.zw;
             // disable depth clipping
-            gl_Position.z = 0.0;
+            // gl_Position.z = 0.0;
```
Comment (willeastcott) [inline src/extras/gizmo/axis-shapes.js:54]: Delete?

File: examples/src/examples/misc/gizmos.example.mjs [hunk 2]
```diff
@@ -122,7 +122,8 @@ data.set('camera', {
 });
 const camera = new pc.Entity('camera');
 camera.addComponent('camera', {
-    clearColor: new pc.Color(0.1, 0.1, 0.1)
+    clearColor: new pc.Color(0.1, 0.1, 0.1),
+    farClip: 1000
```
Comment (willeastcott) [inline examples/src/examples/misc/gizmos.example.mjs:126]: Why won't the example work with the default far clip?

Comment (kpal81xd) [reply examples/src/examples/misc/gizmos.example.mjs:126]: It does it you can just see the the axis lines clipping when zooming out

File: src/extras/gizmo/axis-shapes.js (hunk 1 above)
Comment (kpal81xd) [reply src/extras/gizmo/axis-shapes.js:54]: It looks cleaner when clipping with this so would prefer to have it in but the selection area and rendered area do not align

File: src/extras/gizmo/rotate-gizmo.js [hunk 3]
```diff
@@ -264,28 +265,40 @@ class RotateGizmo extends TransformGizmo {
         return this._shapes.x.tolerance;
     }
 
+    /**
+     * @param {string} prop - The property.
+     * @param {any} value - The value.
```
Comment (willeastcott) [inline src/extras/gizmo/rotate-gizmo.js:270]: Isn't the type here `number`?

File: src/extras/gizmo/scale-gizmo.js [hunk 4]
```diff
@@ -315,40 +315,67 @@ class ScaleGizmo extends TransformGizmo {
         return this._shapes.xyz.tolerance;
     }
 
+    /**
+     * @param {string} prop - The property name.
+     * @param {any} value - The property value.
```
Comment (willeastcott) [inline src/extras/gizmo/scale-gizmo.js:320]: `number`?

File: src/extras/gizmo/scale-gizmo.js [hunk 5]
```diff
@@ -315,40 +315,67 @@ class ScaleGizmo extends TransformGizmo {
...
+     */
     _setArrowProp(prop, value) {
         this._shapes.x[prop] = value;
         this._shapes.y[prop] = value;
         this._shapes.z[prop] = value;
     }
 
+    /**
+     * @param {string} prop - The property name.
+     * @param {any} value - The property value.
```
Comment (willeastcott) [inline src/extras/gizmo/scale-gizmo.js:331]: `number`?

File: src/extras/gizmo/transform-gizmo.js [hunk 6]
```diff
@@ -480,12 +485,22 @@ class TransformGizmo extends Gizmo {
...
+     */
     _colorSemi(color) {
         const clone = color.clone();
         clone.a = this._colorAlpha;
         return clone;
     }
 
+    /**
+     * @param {string} axis - The axis to update.
+     * @param {any} value - The value to set.
```
Comment (willeastcott) [inline src/extras/gizmo/transform-gizmo.js:501]: `Color`?

File: src/extras/gizmo/translate-gizmo.js [hunk 7]
```diff
@@ -305,18 +311,31 @@ class TranslateGizmo extends TransformGizmo {
         return this._shapes.face.tolerance;
     }
 
+    /**
+     * @param {string} prop - The property to set.
+     * @param {any} value - The value to set.
```
Comment (willeastcott) [inline src/extras/gizmo/translate-gizmo.js:316]: OK, I'm gonna stop pointing these out now....maybe there's a good reason it's `any`! 😄

**Review by willeastcott — APPROVED** [review]

File: src/extras/gizmo/translate-gizmo.js (hunk 7 above)
Comment (kpal81xd) [reply src/extras/gizmo/translate-gizmo.js:316]: All of these are any because its a generic property update so the property could technically be any type

**Pull request merged**